### Want to use this script to see the amount of images in each path
###### Helpful for poster :)

### Lets check how many images are in snake vs no snake before sorting

In [ ]:
import os

count = 0
dir_path = r'C:\Build\SnakeImages\NoSnake'
for path in os.scandir(dir_path):
    if path.is_file():
        count += 1
print('file count:', count)

In [ ]:
count = 0
dir_path = r'C:\Build\SnakeImages\Snake'
for path in os.scandir(dir_path):
    if path.is_file():
        count += 1
print('file count:', count)

### Now after sorting, how is the spread? Want it to be as even as possible!
##### ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

In [1]:
# first up is SnakeImages -> test -> no snake
import os

count = 0
dir_path = r'C:\Build\SnakeImages\test\NoSnake'
for path in os.scandir(dir_path):
    if path.is_file():
        count += 1
print('file count:', count)

file count: 147


In [2]:
# next up is SnakeImages -> test -> snake
count = 0
dir_path = r'C:\Build\SnakeImages\test\Snake'
for path in os.scandir(dir_path):
    if path.is_file():
        count += 1
print('file count:', count)

file count: 53


In [3]:
# next up is SnakeImages -> train -> no snake
count = 0
dir_path = r'C:\Build\SnakeImages\train\NoSnake'
for path in os.scandir(dir_path):
    if path.is_file():
        count += 1
print('file count:', count)

file count: 708


In [4]:
# next up is SnakeImages -> train -> snake
count = 0
dir_path = r'C:\Build\SnakeImages\train\Snake'
for path in os.scandir(dir_path):
    if path.is_file():
        count += 1
print('file count:', count)

file count: 265


In [5]:
# next up is SnakeImages -> validate -> no snake
count = 0
dir_path = r'C:\Build\SnakeImages\validate\NoSnake'
for path in os.scandir(dir_path):
    if path.is_file():
        count += 1
print('file count:', count)

file count: 147


In [6]:
# next up is SnakeImages -> validate -> snake
count = 0
dir_path = r'C:\Build\SnakeImages\validate\Snake'
for path in os.scandir(dir_path):
    if path.is_file():
        count += 1
print('file count:', count)

file count: 53


#### It is clear there is not enough training data containing snakes.
###### Now lets look and see if it is because we are lacking in the original directories. How many total images are there? Does ir correlate with what we see here or should I redo the splitting of the data?

In [7]:
# First lets take a look at how many images are in the largen & spawls VI
count = 0
dir_path = r'C:\Users\olivi\Desktop\SpringResearch\extracted_imagesLSVI_PNGS'
for path in os.scandir(dir_path):
    if path.is_file():
        count += 1
print('file count:', count)

file count: 108


In [8]:
# Now lets look at how many are in the Largen & Spawls VII
count = 0
dir_path = r'C:\Users\olivi\Desktop\SpringResearch\extracted_imagesLSVII_PNGS'
for path in os.scandir(dir_path):
    if path.is_file():
        count += 1
print('file count:', count)

file count: 96


In [9]:
# Finally we will look at the 2023 edition: (remember take 2 had non-inverted colors!)
count = 0
dir_path = r'C:\Users\olivi\Desktop\SpringResearch\Extracted_imagesLS2023_take_two'
for path in os.scandir(dir_path):
    if path.is_file():
        count += 1
print('file count:', count)

file count: 1213


### So, in total there are 1417 images to choose from. The highest number of images in one folder is 708 (train\NoSnake) and the total used is: 1373 images.
###### Redo splitting of data? Or add more?